patches/lua-tree.patch

```diff
@@ -1099,3 +1074,159 @@ index 0121ba715..fc27a9b5e 100644
+-            else
+-                return sock
+-            end
+-        end
+-    end
+-    return nil, err
+-end
+-
+-_M.try = _M.newtry()
+-
+-function _M.choose(table)
+-    return function(name, opt1, opt2)
+-        if base.type(name) ~= "string" then
+-            name, opt1, opt2 = "default", name, opt1
+-        end
+-        local f = table[name or "nil"]
+-        if not f then base.error("unknown key (".. base.tostring(name) ..")", 3)
+-        else return f(opt1, opt2) end
+-    end
+-end
+-
+------------------------------------------------------------------------------
+--- Socket sources and sinks, conforming to LTN12
+------------------------------------------------------------------------------
+--- create namespaces inside LuaSocket namespace
+-local sourcet, sinkt = {}, {}
+-_M.sourcet = sourcet
+-_M.sinkt = sinkt
+-
+-_M.BLOCKSIZE = 2048
+-
+-sinkt["close-when-done"] = function(sock)
+-    return base.setmetatable({
+-        getfd = function() return sock:getfd() end,
+-        dirty = function() return sock:dirty() end
+-    }, {
+-        __call = function(self, chunk, err)
+-            if not chunk then
+-                sock:close()
+-                return 1
+-            else return sock:send(chunk) end
+-        end
+-    })
+-end
+-
+-sinkt["keep-open"] = function(sock)
+-    return base.setmetatable({
+-        getfd = function() return sock:getfd() end,
+-        dirty = function() return sock:dirty() end
+-    }, {
+-        __call = function(self, chunk, err)
+-            if chunk then return sock:send(chunk)
+-            else return 1 end
+-        end
+-    })
+-end
+-
+-sinkt["default"] = sinkt["keep-open"]
+-
+-_M.sink = _M.choose(sinkt)
+-
+-sourcet["by-length"] = function(sock, length)
+-    return base.setmetatable({
+-        getfd = function() return sock:getfd() end,
+-        dirty = function() return sock:dirty() end
+-    }, {
+-        __call = function()
+-            if length <= 0 then return nil end
+-            local size = math.min(socket.BLOCKSIZE, length)
+-            local chunk, err = sock:receive(size)
+-            if err then return nil, err end
+-            length = length - string.len(chunk)
+-            return chunk
+-        end
+-    })
+-end
+-
+-sourcet["until-closed"] = function(sock)
+-    local done
+-    return base.setmetatable({
+-        getfd = function() return sock:getfd() end,
+-        dirty = function() return sock:dirty() end
+-    }, {
+-        __call = function()
+-            if done then return nil end
+-            local chunk, err, partial = sock:receive(socket.BLOCKSIZE)
+-            if not err then return chunk
+-            elseif err == "closed" then
+-                sock:close()
+-                done = 1
+-                return partial
+-            else return nil, err end
+-        end
+-    })
+-end
+-
+-
+-sourcet["default"] = sourcet["until-closed"]
+-
+-_M.source = _M.choose(sourcet)
+-
+-return _M
++return require("patched.url")
+
```

Comment:
I haven't done so yet, but did you ensure the `lua-tree.patch` is still working correctly?